{{cookiecutter.repo_name}}/src/{{cookiecutter.repo_name_slug}}/{{cookiecutter.conversion_name}}/{{cookiecutter.conversion_name}}_convert_all_sessions.py

```diff
@@ -0,0 +1,120 @@
+"""Primary script to run to convert all sessions in a dataset using session_to_nwb."""
+from pathlib import Path
+from typing import Union
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from pprint import pformat
+import traceback
+from tqdm import tqdm
+
+from .{{cookiecutter.conversion_name}}_convert_session import session_to_nwb
+
+
+def dataset_to_nwb(
+    *,
+    data_dir_path: Union[str, Path],
+    output_dir_path: Union[str, Path],
+    max_workers: int = 1,
+    stub_test: bool = False,
+    verbose: bool = True,
+):
+    """Convert the entire dataset to NWB.
+
+    Parameters
+    ----------
+    data_dir_path : Union[str, Path]
+        The path to the directory containing the raw data.
+    output_dir_path : Union[str, Path]
+        The path to the directory where the NWB files will be saved.
+    max_workers : int, optional
+        The number of workers to use for parallel processing, by default 1
+    stub_test : bool, optional
+        Whether to run a stub test, by default False
+    verbose : bool, optional
+        Whether to print verbose output, by default True
+    """
+    data_dir_path = Path(data_dir_path)
+    session_to_nwb_kwargs_per_session = get_session_to_nwb_kwargs_per_session(
+        data_dir_path=data_dir_path,
+        output_dir_path=output_dir_path,
+        stub_test=stub_test,
+        verbose=verbose,
+    )
+
+    futures = []
+    with ProcessPoolExecutor(max_workers=max_workers) as executor:
+        for session_to_nwb_kwargs in session_to_nwb_kwargs_per_session:
+            exception_file_path = data_dir_path / f"ERROR_<nwbfile_name>.txt" # Add error file path here
```

Comment:
Can you expand on this? I feel just having the extension here is enough but maybe you came up with a case where it is better to have it as a separate text. Just want to know more of your workflow as I have never used this.

Reply:
What do you mean by extension?

It's useful to handle all the errors when converting the whole dataset so that a long run doesn't get interrupted by one error halfway through.  Then, I usually inspect each of the error messages and fix them one at a time.

Reply:
Also, print statements can get a little wonky with the parallel processing, so saving the error to a separate text file solves that problem.

Reply:
Typo, I meant exception. Sorry.

Yeah, in my workflow I use a logger for this kind of thing. Let me think a bit more.

Reply:
Maybe a logger is the better tool, I just haven't taken the time to figure those out yet. A .txt file is simple and does the job for me, but if you want to replace it with a logger, just make a suggested commit and I can try it out.

Reply:
I don't want to hold the other additions for this. Let's drop the loggin and merge the rest of the things. Then we can open another PR to discuss and refine this specifically. I don't want to add something that I will have to erase and I feel less confident about this.